crates/cairo-lint-core/src/fix/import_fixes.rs

```diff
@@ -0,0 +1,245 @@
+//! 2. Creating import fixes: Generate `ImportFix` structures for each unused import.
+//! 3. Applying fixes: Remove or modify the imports based on the collected fixes.
+//!
+//! The module handles both single imports and multi-imports, ensuring that only unused
+//! items are removed while preserving the structure of the import statements.
+
+use std::collections::HashMap;
+
+use cairo_lang_compiler::db::RootDatabase;
+use cairo_lang_defs::ids::UseId;
+use cairo_lang_diagnostics::DiagnosticEntry;
+use cairo_lang_filesystem::ids::FileId;
+use cairo_lang_semantic::diagnostic::SemanticDiagnosticKind;
+use cairo_lang_semantic::SemanticDiagnostic;
+use cairo_lang_syntax::node::db::SyntaxGroup;
+use cairo_lang_syntax::node::kind::SyntaxKind;
+use cairo_lang_syntax::node::{SyntaxNode, TypedStablePtr, TypedSyntaxNode};
+use cairo_lang_utils::Upcast;
+
+/// Represents a fix for unused imports in a specific syntax node.
+#[derive(Debug, Clone)]
+pub struct ImportFix {
+    /// The node that contains the imports to be fixed.
+    pub node: SyntaxNode,
+    /// The items to remove from the imports.
+    pub items_to_remove: Vec<String>,
+}
+
+impl ImportFix {
+    /// Creates a new `ImportFix` for the given syntax node.
+    pub fn new(node: SyntaxNode) -> Self {
+        ImportFix { node, items_to_remove: vec![] }
+    }
+}
+
+use crate::fix::Fix;
+
+/// Collects unused imports from semantic diagnostics.
+///
+/// # Arguments
+///
+/// * `db` - The root database containing the project information.
+/// * `diags` - A vector of semantic diagnostics.
+///
+/// # Returns
+///
+/// A HashMap where keys are FileIds and values are HashMaps of SyntaxNodes to ImportFixes.
+pub fn collect_unused_imports(
+    db: &RootDatabase,
+    diags: &Vec<SemanticDiagnostic>,
+) -> HashMap<FileId, HashMap<SyntaxNode, ImportFix>> {
+    let mut file_fixes = HashMap::new();
+
+    for diag in diags {
+        if let SemanticDiagnosticKind::UnusedImport(id) = &diag.kind {
+            let file_id = diag.location(db.upcast()).file_id;
+
+            let local_fixes = file_fixes.entry(file_id).or_insert_with(HashMap::new);
+            process_unused_import(db, id, local_fixes);
+        }
+    }
+
+    file_fixes
+}
+
+/// Processes an unused import and updates the fixes HashMap.
+///
+/// # Arguments
+///
+/// * `db` - The root database containing the project information.
+/// * `id` - The UseId of the unused import.
+/// * `fixes` - A mutable reference to the HashMap of fixes.
+fn process_unused_import(db: &RootDatabase, id: &UseId, fixes: &mut HashMap<SyntaxNode, ImportFix>) {
+    let unused_node = id.stable_ptr(db).lookup(db.upcast()).as_syntax_node();
+    let mut current_node = unused_node.clone();
+
+    while let Some(parent) = current_node.parent() {
+        match parent.kind(db) {
+            SyntaxKind::UsePathMulti => {
+                fixes
+                    .entry(parent.clone())
+                    .or_insert_with(|| ImportFix::new(parent.clone()))
```

Comment:
working with refs would add a lot of complexity or not ?

Reply:
I don't know how to do it

Reply:
i'll check when i'm back, let's merge this as is